File: 00005_Bailar en la cueva_ tipado estático/default.java
interface Iluminador {
  void iluminar();
}

class Linterna implements Iluminador {
  public void iluminar() {
    // ...ilumina como un haz de luz potente...
  }
} 

class Vela implements Iluminador {
  void alumbrar() {
    // ...ilumina con un resplandor débil...
  }
} 

class Cueva {
  void entrarCon(Iluminador iluminador) {
    iluminador.iluminar();
    proyectarSombrasTerrorificas();
  }
  void proyectarSombrasTerrorificas() {
    // ...    
  }
}

class VisitaAJava {
  void realizar() {
    Cueva laCueva = new Cueva();
    laCueva.entrarCon(new Vela());
  }
} 

File: 00004_Bailar en la cueva_ tipado dinámico/default.rb
class Linterna 
  def iluminar!
    # ...ilumina como un haz de luz potente...
  end
end

class Vela 
  def alumbrar!
    # ...ilumina con un resplandor débil...
  end
end

class Cueva
  def entrar_con!(iluminador) 
    iluminador.iluminar!
    proyectar_sombras_terrorificas!
  end
  def proyectar_sombras_terrorificas!
    #...
  end
end

class VisitaARuby 
  def realizar!
    la_cueva = Cueva.new 
    la_cueva.entrar_con! Vela.new
  end
end